src/app/functions/server/getRelevantGuidedSteps.ts

```diff
@@ -186,23 +154,24 @@ function isEligibleFor(data: InputData, stepId: StepLink["id"]): boolean {
   return false as never;
 }
 
-function hasCompleted(data: InputData, stepId: StepLink["id"]): boolean {
-  if (stepId === "StartScan") {
-    return data.latestScanData?.scan !== null;
-  }
-
-  if (stepId === "ScanResult") {
-    return (
-      data.latestScanData !== null &&
-      Array.isArray(data.latestScanData?.results) &&
-      !data.latestScanData.results.some((scanResult) => {
-        return scanResult.status === "new" && !scanResult.manually_resolved;
-      }) &&
-      data.latestScanData?.scan?.onerep_scan_status === "finished"
-    );
+function hasCompleted(
+  data: StepDeterminationData,
+  stepId: StepLink["id"]
+): boolean {
+  if (stepId === "Scan") {
+    const hasRunScan =
+      typeof data.latestScanData?.scan === "object" &&
+      data.latestScanData?.scan !== null;
+    const hasResolvedAllScanResults =
+      data.latestScanData?.scan?.onerep_scan_status === "finished" &&
+      data.latestScanData.results.every(
+        (scanResult) =>
+          scanResult.manually_resolved || scanResult.status !== "new"
+      );
+    return hasRunScan && hasResolvedAllScanResults;
   }
 
-  function isResolved(
+  function isBreachResolved(
```

Comment:
With the distinction of breaches (HIBP breaches) and data broker results we started using “exposures” as an umbrella term in our codebase. Should we name this `isExposureResolved` or is this meant to be specific to breaches?

Reply:
It's specific to breaches, that's why I renamed it :)